### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by AKHIL-882**

### CODE REVIEW

File: CoinDock-api/app/Http/Controllers/Auth/UserController.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+class UserController extends Controller
+{
+    //
+
+    // protected function generateAccessToken($user){
+    //     $token= $user->createToken($user->email. '-'.now());
+    //     return $token->accessToken;
+    // }
+
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Controllers/Auth/UserController.php]: Can you remove the unused code?

File: CoinDock-api/app/Http/Controllers/Auth/UserController.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            'last_name' => $request->last_name,
+            'user_type' => $request->user_type,
+            'date_of_birth' => $request->date_of_birth,
+            'country' => $request->country,
+            'email' => $request->email,
+            'password' => bcrypt($request->password),
+            're_enter_password' =>bcrypt($request->re_enter_password),
+            'status'=> $request->status
+        ]);
+
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Controllers/Auth/UserController.php]: Can you move the above creation part to the user model?

File: CoinDock-api/app/Http/Controllers/Auth/UserController.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    // protected function generateAccessToken($user){
+    //     $token= $user->createToken($user->email. '-'.now());
+    //     return $token->accessToken;
+    // }
+
+     /**
+     * Registration
+     */
+    public function register(SignupRequest $request)
+    {
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Controllers/Auth/UserController.php]: Can you change the function name to store() or sign()? It's not the register right.

File: CoinDock-api/app/Http/Controllers/Auth/UserController.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * Login
+     */
+    
+    public function login(Request $request)
+    {
+        $data = [
+            'email' => $request->email,
+            'password' => $request->password
+        ];
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Controllers/Auth/UserController.php]: Move this to Request class

File: CoinDock-api/app/Http/Controllers/Auth/UserController.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            'date_of_birth' => $request->date_of_birth,
+            'country' => $request->country,
+            'email' => $request->email,
+            'password' => bcrypt($request->password),
+            're_enter_password' =>bcrypt($request->re_enter_password),
+            'status'=> $request->status
+        ]);
+       
+        $token = $user->createToken('Laravel')->accessToken;
+        return response()->json(["status" => "success", "error" => false, "message" => "Success! User registered."], 201);
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Controllers/Auth/UserController.php]: Please, use the same unique response for all functions.

File: CoinDock-api/app/Http/Requests/V1/SignupRequest.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [
+            'first_name' => 'required|string|max:45',
+            'last_name' => 'required|string|max:45',
+            // 'user_type' => 'required|string',
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Requests/V1/SignupRequest.php]: Please, Remove unused code.

File: CoinDock-api/app/Http/Requests/V1/SignupRequest.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            'date_of_birth' => 'required|date|before_or_equal:'.\Carbon\Carbon::now()->subYears(15)->format('Y-m-d'),
+            'country' => 'required|string',
+            'email' => 'required|email|unique:users',
+            'password' => 'required|max:12',
+            're_enter_password' => 'required|max:12|same:password',
+            //'password' => 'required|max:12|min:4|regex:/^.*(?=.{3,})(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[\d\x])(?=.*[!$#%]).*$/| confirmed',
+             're_enter_password' => 'required|same:password',
```
Comment (sita7-debug) [inline CoinDock-api/app/Http/Requests/V1/SignupRequest.php]: Here, re_enter_password had duplicates.

**Review by sita7-debug — CHANGES_REQUESTED** [review]

**Review by sita7-debug — APPROVED** [review]

**Review by bharatjarvis — APPROVED** [review]

**Pull request merged**